src/query/functions-v2/src/scalars/control.rs

```diff
@@ -140,4 +150,58 @@ pub fn register(registry: &mut FunctionRegistry) {
             }),
         }))
     });
+    registry.register_1_arg_core::<NullableType<GenericType<0>>, BooleanType, _, _>(
+        IS_NULL,
+        FunctionProperty::default(),
+        |_| None,
+        |arg, _| match &arg {
+            ValueRef::Column(NullableColumn { validity, .. }) => {
+                let bitmap = Bitmap::from_iter(validity.iter().map(|v| !v));
+                Ok(Value::Column(bitmap))
+            }
+            ValueRef::Scalar(None) => Ok(Value::Scalar(true)),
+            ValueRef::Scalar(Some(_)) => Ok(Value::Scalar(false)),
+        },
+    );
+    registry.register_1_arg_core::<NullType, BooleanType, _, _>(
```

Comment:
Null should be registered before Nullable<Generic> because Null could be automatically cast to Nullable and therefore has no chance to be matched.

Reply:
I don't think so. The way it's written now, NullType will match.